pwiz_tools/Skyline/Model/TransitionGroup.cs

```diff
@@ -726,14 +726,20 @@ protected override IList<ExplicitLoss> CreateState(ExplicitLoss[] mods)
                 // there should be only one loss
                 TransitionLosses firstLosses = null;
                 List<TransitionLosses> allLosses = null;
-                HashSet<double> allLossMasses = null;
+                HashSet<Tuple<double, int>> allLossMassCharges = null;
                 foreach (var losses in potentialLosses)
                 {
-                    double lossMass = CalcTransitionLossesMass(type, cleavageOffset, massType, losses);
+                    double lossMass = CalcTransitionLossesMass(type, cleavageOffset, massType, losses, out var lossCharge);
+                    if (type != IonType.precursor)
+                        lossCharge = 0; // Only precursors care about the loss charge

```

Comment:
I don't understand this comment. Fragments do sort of care about the loss charge in that the fragment charge has to be less than or equal to the precursor's charge minus the loss charge.

Reply:
Until I added this the case of interest was adding multiple equivalent losses to y- and b-ions.

e.g. H20, H20[M-1], H20[M-2] on a triply charged precursor could all lead to:

y7-16+ (neutral loss from the y7+)
y7-16+ (-1 loss from the y7++)
y7-16+ (-2 loss from the y7+++)

Without this condition all 3 got added to the list of possible transitions as if they were different transitions. The case of whether a loss can be applied without reducing the charge to zero is handled elsewhere.

Reply:
I mean -18 of course for water loss.